src/main/scala/Column.scala

```diff
@@ -1034,13 +1096,88 @@ case class Column private (
     }
   }
 
+  def validateFormat(value: String, rowNumber: Long): Boolean = {
+    format match {
+      case Some(f) => {
+        val formatValidator = datatypeFormatValidation(baseDataType)
+        if (formatValidator(value)) true
+        else {
+          errors = errors :+ ErrorWithCsvContext(
+            "format",
+            "schema",
+            rowNumber.toString,
+            columnOrdinal.toString,
+            value,
+            s"Value in csv does not match the format specified in metadata, Value: ${value} Format: ${f.pattern.get}"
+          )
+          false
+        }
+      }
+      case None => true
+    }
+  }
+
+  def validateLength(value: String, rowNumber: Long): Boolean = {
+    var valid = true
+    if (
+      datatype.path("length").isMissingNode && datatype
+        .path("minLength")
+        .isMissingNode && datatype.path("maxLength").isMissingNode
+    ) {
+      valid = true
+    } else {
+      var length = value.length
+      if (baseDataType == "http://www.w3.org/2001/XMLSchema#base64Binary") {
+        length = value.replaceAll("==?$", "").length * (3 / 4)
+      } else if (baseDataType == "http://www.w3.org/2001/XMLSchema#hexBinary") {
+        length = value.length / 2
+      }
+      if (minLength.isDefined && length < minLength.get) {
+        errors = errors :+ ErrorWithCsvContext(
+          "minLength",
+          "schema",
+          rowNumber.toString,
+          columnOrdinal.toString,
+          value,
+          s"value length less than minLength specified - $minLength"
+        )
+        valid = false
+      }
+      if (maxLength.isDefined && length > maxLength.get) {
+        errors = errors :+ ErrorWithCsvContext(
+          "maxLength",
+          "schema",
+          rowNumber.toString,
```

Comment:
Again, I'm not particularly happy with the column being told its row number. Feels like we could add the context in whatever method calls this one.

Reply:
Done